packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -192,6 +192,27 @@ class DatePicker extends React.Component {
     this.isDefaultDateAcceptable = this.validateDefaultDate();
   }
 
+  getMetadata() {
+    const format = DateUtil.getFormatByLocale(this.props.intl.locale);
+    const isCompleteDate = DateUtil.isValidDate(this.dateValue, format);
+    const iSOString = isCompleteDate ? DateUtil.convertToISO8601(this.dateValue, format) : '';
+
+    let isValidDate = false;
+
+    if (this.dateValue === '' || (isCompleteDate && this.isDateWithinRange(DateUtil.createSafeDate(iSOString)))) {
```

Comment:
this can be simplified to   
```
 if (this.dateValue === '' || this.isDateWithinRange(DateUtil.createSafeDate(iSOString)))
```

Reply:
The check for `isCompleteDate` is required because, if the date is incomplete then the `isoString` will be undefined (which may cause an exception).